### PR TITLE
Update to work with ink! 5.0.0

### PR DESCRIPTION
**Pull request by h4nsu**

### CODE REVIEW

File: events.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+#[ink::event]
+pub struct Transfer {
+    #[ink(topic)]
+    pub from: Option<AccountId>,
+    #[ink(topic)]
+    pub to: Option<AccountId>,
+    pub value: u128,
+}
```
Comment (deuszx) [inline events.rs:19]: Can you add comments on these public-facing structs? It will be useful to explain what `None`s mean.

Comment (h4nsu) [reply events.rs]: Good idea, done.

**Comment by deuszx** [comment]
In general I'm ready to approve the PR but I wonder whether you've considered the current situation with 5.0.0-rc release - the "rc" part means it's release candidate only, so there will come 5.0.0 release at some point. In theory, 5.0.0 release can be backwards incompatible with the *-rc release and here we're already releasing the new version. 

What do you think?

**Comment by h4nsu** [comment]
Yes, I'm aware it's maybe a bit premature. But I tested these changes also on version https://github.com/paritytech/ink/pull/2063 which most likely will end up as ink! 5.0.0. Also, this repo here has versioned releases to crates, I think nothing particularly bad will happen if the `main` branch contains some "work-in-progress" for some time.

**Comment by deuszx** [comment]
So just by bumping the version in `Cargo.toml` here, it won't publish automatically to crates.io?

**Comment by h4nsu** [comment]
Nope, publishing on crates.io is a manual process

**Review by deuszx — APPROVED** [review]

**Comment by deuszx** [comment]
> Nope, publishing on crates.io is a manual process triggered locally via `cargo`, not from GH

Personally I prefer when it's done automatically, when it detects a new version of `Cargo.toml` but that's for another discussion.

**Review by UladAleph — APPROVED** [review]
Looks cool, I've been playing around with it and everything seems to work, nicely done!

**Pull request merged**